node/src/effect/console.rs

```diff
@@ -0,0 +1,60 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Display},
+};
+
+use casper_types::EraId;
+use datasize::DataSize;
+use futures::Future;
+use serde::Serialize;
+
+use super::Responder;
+use crate::components::consensus::EraDump;
+
+/// A request to dump the internal consensus state of a specific era.
+#[derive(DataSize, Serialize)]
+pub(crate) struct DumpConsensusStateRequest {
+    /// Era to serialize.
+    ///
+    /// If not given, use active era.
+    pub(crate) era_id: Option<EraId>,
+    /// Serialization function to serialize the given era with.
+    #[data_size(skip)]
+    #[serde(skip)]
+    pub(crate) serialize: fn(&EraDump) -> Result<Vec<u8>, Cow<'static, str>>,
+    /// Responder to send the serialized representation into.
+    pub(crate) responder: Responder<Result<Vec<u8>, Cow<'static, str>>>,
```

Comment:
Looks like neither this `pub(crate)`, nor the one on `serialize` are necessary.

Reply:
`rustc` seems to differ :)

```
error[E0451]: field `serialize` of struct `DumpConsensusStateRequest` is private
    --> node/src/effect.rs:1829:17
     |
1829 |                 serialize,
     |                 ^^^^^^^^^ private field

error[E0451]: field `responder` of struct `DumpConsensusStateRequest` is private
    --> node/src/effect.rs:1830:17
     |
1830 |                 responder,
     |                 ^^^^^^^^^ private field
```

Reply:
Oh interesting. It passed `cargo check` for me.